### PR TITLE
Add filters and advanced filters options

### PR DESCRIPTION
**Pull request by subinasr**
- Addresses #57 #60 #59 #62 
- Depends on server branch fix/combined-indicator-options

## Changes

*  Add filter options
* Add a clear button to clear the filters
* Add advanced filter options
* Keywords list pending (last minute changes)

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

**Review by shreeyash07 — APPROVED** [review]

**Review by priyesh777 — APPROVED** [review]

File: app/views/Dashboard/AdvancedFilters/index.tsx [hunk 1]
```diff
@@ -16,105 +26,50 @@ export interface AdvancedOptionType {
...
 
-interface FilterType {
+const THEMATICS = gql`
+    query Thematics($type: String!) {
+        filterOptions {
+            thematics(type: $type)
+        }
+    }
+`;
+
```
Comment (AdityaKhatri) [inline app/views/Dashboard/AdvancedFilters/index.tsx]: Why not pull this together with other gql queries that run on mount?
Lets!

File: app/views/Dashboard/AdvancedFilters/index.tsx [hunk 2]
```diff
@@ -16,105 +26,50 @@ export interface AdvancedOptionType {
...
+interface Thematic {
     key: string;
     label: string;
 }
```
Comment (AdityaKhatri) [inline app/views/Dashboard/AdvancedFilters/index.tsx]: Use generic or from the gql query?

File: app/views/Dashboard/AdvancedFilters/index.tsx [hunk 3]
```diff
@@ -179,6 +134,56 @@ function AdvancedFilters(props: Props) {
...
     );
 
+    const {
+        data: typeList,
+        loading: typesLoading,
+    } = useQuery<TypesQuery, TypesQueryVariables>(
+        TYPES,
+    );
+
+    const types = typeList?.filterOptions?.types.map((t) => ({
```
Comment (AdityaKhatri) [inline app/views/Dashboard/AdvancedFilters/index.tsx]: what is t? Please write full name plz

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/Dashboard/Filters/index.tsx [hunk 4]
```diff
@@ -200,6 +228,29 @@ function Filters(props: Props) {
...
+        SUBVARIABLES,
+        {
+            variables: subvariablesVariables,
+        },
+    );
+
+    const subvariables = subvariableList?.filterOptions?.subvariables.map((sub) => ({
+        key: sub,
+        label: sub,
+    }));
```
Comment (AdityaKhatri) [inline app/views/Dashboard/Filters/index.tsx]: use use memo

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

**Pull request merged**